### PR TITLE
add reusable modal

### LINKED ISSUE #199: Add Modal View
Add a package or component that allows showing it's children as a modal on the top level

### PR DESCRIPTION
**Pull request by DasProffi**
closes #199

### CODE REVIEW

**Review by use-to — CHANGES_REQUESTED** [review]
1. modal overlay - click event (dismiss function) should be optional
2. modal overlay - background should be darker
3. fix `onClick`-event on actual modal container (currently triggers `onClose`)

File: tasks/components/modals/Modal.tsx
Comment (JannikWibker) [inline tasks/components/modals/Modal.tsx:1]: could also be moved to `lib/`

File: tasks/components/modals/ConfirmDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        <Button color="accent" onClick={event => {
+          event.stopPropagation()
+          onClose('closed')
+        }}>
+          {translation.cancel}
+        </Button>
+        {isShowingDecline && (
+          <Button color="negative" onClick={event => {
+            event.stopPropagation()
+            onClose('decline')
```
Comment (JannikWibker) [inline tasks/components/modals/ConfirmDialog.tsx:60]: I'd vouch for having individual callbacks for each action (`'onDecline'`, `onAccept` or `onConfirm`, `onClose`)

But I'm open to discussing this; if there is something specific this allows I'm all for it

File: tasks/components/modals/Modal.tsx
Comment (use-to) [reply tasks/components/modals/Modal.tsx:1]: I would say there are many components we could move to the `lib/` directory.
Maybe we should do it later on?

Comment (DasProffi) [reply tasks/components/modals/Modal.tsx:1]: I will make an issue for that #212

**Review by use-to — APPROVED** [review]

**Pull request merged**